curriculum/challenges/english/20-upcoming-python/learn-recursion-by-solving-the-tower-of-hanoi-puzzle/64e34146860065146733883b.md

```diff
@@ -0,0 +1,76 @@
+---
+id: 64e34146860065146733883b
+title: Step 53
+challengeType: 20
+dashedName: step-53
+---
+
+# --description--
+
+Although recursion could sometimes be less easy to understand, it gives you the power to create more concise code. In this case, you don't even need to differentiate between even and odd numbers of disks. As a final step, set `NUMBER_OF_DISKS` to `5` and check the output.
+
+Well done. You have completed the Tower of Hanoi practice project.
+
+# --hints--
+
+You should set the `NUMBER_OF_DISKS` variable to 5.
+
+```js
+({test: () => assert.equal(__userGlobals.get('NUMBER_OF_DISKS'), 5) })
+```
+
+# --seed--
+
+## --seed-contents--
+
+```py
+--fcc-editable-region--
+NUMBER_OF_DISKS = 4
+A = list(range(NUMBER_OF_DISKS, 0, -1))
+B = []
+C = []
+
+def move(n, source, target, auxiliary):
+    if n > 0:
+        # move n - 1 disks from source to auxiliary, so they are out of the way
+        move(n - 1, source, auxiliary, target)
+        
+        # move the nth disk from source to target
+        target.append(source.pop())
+        
+        # display our progress
+        print(A, B, C, '\n')
+        
+        # move the n - 1 disks that we left on auxiliary onto target
+        move(n - 1, auxiliary, target, source)
+              
+# initiate call from source A to target C with auxiliary B
+move(NUMBER_OF_DISKS, A, C, B)
+--fcc-editable-region--
+```
+
+# --solutions--
+
+```py
+NUMBER_OF_DISKS = 5
+A = list(range(NUMBER_OF_DISKS, 0, -1))
+B = []
+C = []
+
+def move(n, source, target, auxiliary):
+    if n > 0:
+        # move n - 1 disks from source to auxiliary, so they are out of the way
+        move(n - 1, source, auxiliary, target)
+        
+        # move the nth disk from source to target
+        target.append(source.pop())
+        
+        # display our progress
+        print(A, B, C, '\n')
+        
+        # move the n - 1 disks that we left on auxiliary onto target
+        move(n - 1, auxiliary, target, source)
```

Comment:
```suggestion
    if n <= 0:
        return
    # move n - 1 disks from source to auxiliary, so they are out of the way
    move(n - 1, source, auxiliary, target)
    
    # move the nth disk from source to target
    target.append(source.pop())
    
    # display our progress
    print(A, B, C, '\n')
    
    # move the n - 1 disks that we left on auxiliary onto target
    move(n - 1, auxiliary, target, source)
```
Did I suggest this before? It could go as a final touch to the function, after `rods` dictionary is removed.

Reply:
Don't remember if you've already done it. Anyway this is going to be in the next commit 👍